src/Tvheadend.cpp

```diff
@@ -1651,19 +1651,36 @@ void CTvheadend::VfsClose()
 
 ssize_t CTvheadend::VfsRead(unsigned char *buf, unsigned int len)
 {
-  return m_vfs->Read(buf, len);
+  bool inprogress = (m_playingRecording->GetState() == PVR_TIMER_STATE_RECORDING);
+  return m_vfs->Read(buf, len, inprogress);
 }
 
 long long CTvheadend::VfsSeek(long long position, int whence)
 {
-  return m_vfs->Seek(position, whence);
+  bool inprogress = (m_playingRecording->GetState() == PVR_TIMER_STATE_RECORDING);
+  return m_vfs->Seek(position, whence, inprogress);
 }
 
 long long CTvheadend::VfsSize()
 {
   return m_vfs->Size();
 }
 
+void CTvheadend::VfsPauseStream(bool paused)
+{
+  /* only process this event for in-progress recordings */
+  if (m_playingRecording->GetState() == PVR_TIMER_STATE_RECORDING)
+    m_vfs->PauseStream(paused);
+}
+
+bool CTvheadend::VfsIsRealTimeStream()
+{
+  if (m_playingRecording->GetState() == PVR_TIMER_STATE_RECORDING)
```

Comment:
As you are using `m_playingRecording->GetState() == PVR_TIMER_STATE_RECORDING` a lot, you probably should introduce a private inline member function`CTvheadend::VfsIsActiveRecording() const`.

Reply:
Done - New inline VfsIsActiveRecording() now used in each of the 4 places this check was used.

Reply:
Nope, cannot see this change in the latest commit.